### PR TITLE
Add jitter to constant backoff

### LINKED ISSUE #37: Add jitter support for constant back off

### PR DESCRIPTION
**Pull request by balroggg**
- add jitter to `ConstantBackoff`
- change `rand` to `fastrand`
- change `futures` to `futures-core`

Fix #37

### CODE REVIEW

**Comment by balroggg** [comment]
Change dependencies to reduce compilation time a little. Fell free to ask revert some changes.

File: src/constant.rs [hunk 1]
```diff
@@ -146,4 +161,13 @@ mod tests {
...
         assert_eq!(None, exp.next());
     }
+
+    #[test]
+    fn test_constant_with_jitter() {
+        let mut it = ConstantBuilder::default().with_jitter().build();
+
+        let dur = it.next().unwrap();
+        fastrand::seed(7);
+        assert!(dur > Duration::from_secs(1));
```
Comment (balroggg) [inline src/constant.rs:171]: It's hard to test random things, this should be something like `delay <= dur < delay + rng`

File: src/constant.rs [hunk 2]
```diff
@@ -96,14 +107,18 @@ impl Iterator for ConstantBackoff {
     type Item = Duration;
 
     fn next(&mut self) -> Option<Self::Item> {
+        let delay = || match self.jitter {
+            true => self.delay + self.delay.mul_f32(fastrand::f32()),
```
Comment (balroggg) [inline src/constant.rs:111]: In default case it's a little strange to set delay(1s) + jitter(1s in some cases), so double the main delay. But maybe it's ok.

**Review by Xuanwo — APPROVED** [review]
Great!

**Comment by Xuanwo** [comment]
Thanks a lot!

File: src/constant.rs (hunk 2 above)
Comment (Xuanwo) [reply src/constant.rs:111]: We can add jitter range for user in the future, let's merge first.

File: src/constant.rs (hunk 1 above)
Comment (Xuanwo) [reply src/constant.rs:171]: Are you willing to start a new PR for this?

Comment (balroggg) [reply src/constant.rs:171]: Maybe for unit test is enough. I will try to find a better solution, but I'm not sure.

**Pull request merged**